monkey/monkey_island/cc/ui/src/components/ui-components/IslandResetModal.tsx

```diff
@@ -91,15 +89,32 @@ const IslandResetModal = (props: Props) => {
     }
   }
 
-  function resetIsland(url: string, callback: () => void) {
-    auth.authFetch(url)
-      .then(res => res.json())
+  function clearSimulationData(callback: () => void) {
+    auth.authFetch('/api/clear-simulation-data', {method: 'POST'})
       .then(res => {
-        if (res['status'] === 'OK') {
+        if (res.status === 200) {
           callback()
         }
       })
   }
+  function resetAll(callback: () => void) {
```

Comment:
If this returned a prommise we wouldn't need the callback

Reply:
Maybe a try/catch block would be better. Now there's nothing the caller can do if this reset failed

Reply:
I don't know if it's worth actually implementing the error message, but I think it's worth to not pass in the callback so it would be easier later to find this and refactor if needed.